fpr/migrations-data/mc_validate_cmd_2.py

```diff
@@ -0,0 +1,164 @@
+    lxml etree parse of the output.
+    """
+
+    args = ['mediaconch', '-mc', '-fx', '-iv', '4', target]
+    try:
+        output = subprocess.check_output(args)
+    except subprocess.CalledProcessError:
+        raise MediaConchException("MediaConch failed when running: %s" % (
+            ' '.join(args),))
+    try:
+        return Parse(etree_el=etree.fromstring(output), stdout=output)
+    except etree.XMLSyntaxError:
+        raise MediaConchException(
+            "MediaConch failed when attempting to parse the XML output by"
+            " MediaConch")
+
+
+def get_impl_check_name(impl_check_el):
+    name_el = impl_check_el.find('%sname' % NS)
+    if name_el is not None:
+        return name_el.text
+    else:
+        return 'Unnamed Implementation Check %s' % uuid.uuid4()
+
+
+def get_check_name(check_el):
+    return check_el.attrib.get(
+        'name', check_el.attrib.get('icid', 'Unnamed Check %s' % uuid.uuid4()))
+
+
+def get_check_tests_outcomes(check_el):
+    """Return a list of outcome strings for the <check> element `check_el`."""
+    outcomes = []
+    for test_el in check_el.iterfind('%stest' % NS):
+        outcome = test_el.attrib.get('outcome')
+        if outcome:
+            outcomes.append(outcome)
+    return outcomes
+
+
+def get_impl_check_result(impl_check_el):
+    """Return a dict mapping check names to lists of test outcome strings."""
+    checks = {}
+    for check_el in impl_check_el.iterfind('%scheck' % NS):
+        check_name = get_check_name(check_el)
+        test_outcomes = get_check_tests_outcomes(check_el)
+        if test_outcomes:
+            checks[check_name] = test_outcomes
+    return checks
+
+
+def get_impl_checks(doc):
+    """When not provided with a policy file, MediaConch produces a series of
+    XML <implementationChecks> elements that contain <check> sub-elements. This
+    function returns a dict mapping implementation check names to dicts that
+    map individual check names to lists of test outcomes, i.e., 'pass' or
+    'fail'.
+
+    """
+
+    impl_checks = {}
+    path = '.%smedia/%simplementationChecks' % (NS, NS)
+    for impl_check_el in doc.iterfind(path):
+        impl_check_name = get_impl_check_name(impl_check_el)
+        impl_check_result = get_impl_check_result(impl_check_el)
+        if impl_check_result:
+            impl_checks[impl_check_name] = impl_check_result
+    return impl_checks
+
+
+def get_event_outcome_information_detail(impl_checks):
+    """Return a 2-tuple of info and detail.
+
+    - info: 'pass' or 'fail'
+    - detail: human-readable string indicating which implementation checks
+      passed or failed. If implementation check as a whole passed, just return
+      the passed check names; if it failed, just return the failed ones.
+
+    """
+    info = 'pass'
+    failed_impl_checks = []
+    passed_impl_checks = []
+    for impl_check, checks in impl_checks.items():
```

Comment:
@ross-spencer when you run `mediaconch -mc -fx -iv 4 /some/path/to/some.mkv` you get XML with one or more `<implementationChecks>` elements. Each of these has a `<name>` element and zero or more `<check>` elements. Each `<check>` element has an `icid` attribute and contains one or more `<test>` elements with `outcome` attributes having values `'pass'` or `'fail'`. 

So, something like the following:

        <implementationChecks ...>
          <name>MediaConch EBML Implementation Checker</name>
          <check icid="EBML-ELEM-START" ...>
            <test outcome="fail">
              <value .../>
            </test>
          </check>
        </implementationChecks>

would be parsed by `get_impl_checks` to

    {'MediaConch EBML Implementation Checker': {
        'EBML-ELEM-START': ['fail']}}

and the resulting `eventOutcomeDetailNote` would be

    MediaConch implementation check result: The implementation check MediaConch EBML Implementation Checker returned failure for the following check(s): EBML-ELEM-START.

To be honest, I don't really know what these checks mean. @ross-spencer or @ablwr, you two would obviously know about the semantics of these checks. The above is all I really know.

@ross-spencer are you thinking that it would be good for me to add more comments or docstrings (like above) in the FPCommand that explain how we go from MC XML to a Python data structure (dict), and finally to a string for the user to read? That's one question. The other question is Do we need to explain in the code what exactly the MediaConch implementation check is doing and what its checks signify? If the answer to the second question is "yes", I'll need some assistance with that and I would guess we don't want to require this for the 1.7 release.